lib/perl/Genome/InstrumentData/AlignmentResult/Command/RecreatePerLaneBam.pm

```diff
@@ -150,8 +153,17 @@ sub _compare_flagstat {
         die $self->error_message("Fail to run flagstat on $temp_bam");
     }
     
-    unless (compare($temp_flagstat, $flagstat) == 0) {
-        die $self->error_message("The bam flagstat from the extracting is different from the comparison flagstat $flagstat");
+    if ($diff) { #expect only one line diff on duplicates
```

Comment:
Possible alternatives implementations...

1) Might be better because it leaves parsing of the flagstat file, however trivial it may be, in the place designated for parsing flagstat files.

```
use List::MoreUtils qw(all uniq);
...
my %flagstat_data = Genome::Model::Tools::Sam::Flagstat->parse_file_into_hashref($flagstat);
my %temp_flagstat_data = Genome::Model::Tools::Sam::Flagstat->parse_file_into_hashref($temp_flagstat);
delete $flagstat_data{reads_marked_duplicates};
delete $temp_flagstat_data{reads_marked_duplicates};
my @keys = uniq keys(%flagstat_data), keys(%temp_flagstat_data);
unless (all { $flagstat_data{$_} eq $temp_flagstat_data{$_} } @keys) {
    die ...
}
```

2) This technique would allow `_compare_flagstat()` to take `$comparator` as an argument just like `compare()` can.

```
my $comparator = sub { $_[0] !~ /duplicates/ && $_[0] ne $_[1] };
$self->_compare_flagstat($temp_bam, $flagstat, $comparator);
...
sub _compare_flagstat {
    my ($self, $temp_bam, $flagstat, $comparator) = @_;
    ...
    my @compare_args = ($temp_flagstat, $flagstat);
    if ($comparator) {
        push @compare_args, $comparator;
    }
    ...
    unless (compare(@compare_args) == 0) {
        die ...
    }
}
```